pkg/config/dp-server/config.go

```diff
@@ -61,5 +65,79 @@ func DefaultDpServerConfig() *DpServerConfig {
 		Auth: DpServerAuthConfig{
 			Type: "", // autoconfigured from the environment
 		},
+		Hds: DefaultHdsConfig(),
+	}
+}
+
+func DefaultHdsConfig() *HdsConfig {
+	return &HdsConfig{
+		Enabled:         true,
+		Interval:        1 * time.Second,
+		RefreshInterval: 1 * time.Second,
+		Check: &HdsCheck{
+			Timeout:            2 * time.Second,
+			Interval:           1 * time.Second,
+			NoTrafficInterval:  1 * time.Second,
+			HealthyThreshold:   1,
+			UnhealthyThreshold: 1,
+		},
+	}
+}
+
+type HdsConfig struct {
+	// Enabled if true then Envoy will actively check application's ports, but only on Universal.
+	// On Kubernetes this feature disabled for now regardless the flag value
+	Enabled bool `yaml:"enabled" envconfig:"kuma_dp_server_hds_enabled"`
+	// Interval for Envoy to send statuses for HealthChecks
+	Interval time.Duration `yaml:"interval" envconfig:"kuma_dp_server_hds_interval"`
+	// RefreshInterval is an interval for re-genarting configuration for Dataplanes connected to the Control Plane
+	RefreshInterval time.Duration `yaml:"refreshInterval" envconfig:"kuma_dp_server_hds_refresh_interval"`
+	// Check defines a HealthCheck configuration
+	Check *HdsCheck `yaml:"check"`
+}
+
+func (h *HdsConfig) Sanitize() {
+}
+
+func (h *HdsConfig) Validate() error {
+	if h.Interval <= 0 {
+		return errors.New("Interval must be greater than 0s")
 	}
+	if err := h.Check.Validate(); err != nil {
+		return errors.Wrap(err, "Check is invalid")
+	}
+	return nil
+}
+
+type HdsCheck struct {
+	// Timeout is a time to wait for a health check response. If the timeout is reached the
+	// health check attempt will be considered a failure.
+	Timeout time.Duration `yaml:"timeout" envconfig:"kuma_dp_server_hds_check_timeout"`
+	// Interval between health checks.
+	Interval time.Duration `yaml:"interval" envconfig:"kuma_dp_server_hds_check_interval"`
+	// NoTrafficInterval is a special health check interval that is used when a cluster has
+	// never had traffic routed to it.
+	NoTrafficInterval time.Duration `yaml:"noTrafficInterval" envconfig:"kuma_dp_server_hds_check_no_traffic_interval"`
+	// HealthyThreshold is a number of healthy health checks required before a host is marked
+	// healthy.
+	HealthyThreshold uint32 `yaml:"healthyThreshold" envconfig:"kuma_dp_server_hds_check_healthy_threshold"`
+	// UnhealthyThreshold is a number of unhealthy health checks required before a host is marked
+	// unhealthy.
+	UnhealthyThreshold uint32 `yaml:"unhealthyThreshold" envconfig:"kuma_dp_server_hds_check_unhealthy_threshold"`
+}
+
+func (h *HdsCheck) Sanitize() {
+}
+
+func (h *HdsCheck) Validate() error {
+	if h.Timeout <= 0 {
+		return errors.New("Timeout must be greater than 0s")
+	}
+	if h.Interval <= 0 {
+		return errors.New("Interval must be greater than 0s")
+	}
+	if h.NoTrafficInterval <= 0 {
+		return errors.New("NoTrafficInterval must be greater than 0s")
+	}
+	return nil
```

Comment:
Please make sure these are not conflicting. For example, can the timeout be lower than the check interval?